### PR TITLE
Feat: display validation errors

### PR DESCRIPTION
**Pull request by dleard**
colour for error messages / icon taken from:
http://web-accessibility.carnegiemuseums.org/design/color/

### CODE REVIEW

File: app/components/Form/FormBase.tsx [hunk 1]
```diff
@@ -1,24 +1,49 @@
...
       <style jsx>{`
         :global(label) {
           font-weight: bold;
         }
+        .errors {
+          margin: 1em 0 0 0;
+          padding: 20px;
+          background: #ce5c5c;
+          color: white;
+          font-size: 20px;
```
Comment (matthieu-foucault) [inline app/components/Form/FormBase.tsx:38]: I don't think that is does anything, or is the style we want for errors

File: app/lib/theme/customTransformErrors.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+export const customTransformErrors = (
+  errors: AjvError[],
+  customFormatsErrorMessages: { [key: string]: string }
+) => {
+  // Ignore oneOf errors https://github.com/rjsf-team/react-jsonschema-form/issues/1263
+  return errors
+    .filter((error) => error.name !== "oneOf")
+    .map((error) => {
+      if (error.name !== "format") return error;
```
Comment (matthieu-foucault) [inline app/lib/theme/customTransformErrors.ts:11]: Can we update `is a required property` (`error.name == "required"`) to a more actionable phrasing?
e.g. "Please enter a value.". For dropdown and radio buttons, "Please select a value." would be more appropriate, but I'm not sure if we can easily find out the input type here

File: app/lib/theme/FieldTemplate.tsx [hunk 3]
```diff
@@ -1,15 +1,38 @@
...
       </div>
       <style jsx>{`
         div:not(:last-child) {
           margin-bottom: 1em;
         }
+        :global(.error-icon) {
+          color: #cd2026;
+        }
+        :global(.error-div) {
+          height: 5%;
```
Comment (matthieu-foucault) [inline app/lib/theme/FieldTemplate.tsx:32]: should this height be in `rem` instead? A height in `%` will be relative to the parent component's height, no?

File: app/components/Form/FormBase.tsx [hunk 4]
```diff
@@ -1,24 +1,49 @@
...
+        }
+        :global(ul.error-detail) {
+          padding: 0;
+          list-style: none;
+        }
+        :global(li.text-danger) {
+          color: #cd2026 !important;
+        }
```
Comment (matthieu-foucault) [inline app/components/Form/FormBase.tsx:46]: Are those needed? I see styles in the `FieldTemplate` as well

File: app/lib/theme/FieldTemplate.tsx [hunk 5]
```diff
@@ -1,15 +1,38 @@
...
+import { faExclamationTriangle } from "@fortawesome/free-solid-svg-icons";
+
+const FieldTemplate = ({ children, errors, help, rawErrors }) => {
   return (
     <>
       <div>
         {children}
-        {errors}
+        {rawErrors ? (
+          <div className={"error-div"}>
```
Comment (matthieu-foucault) [inline app/lib/theme/FieldTemplate.tsx:10]:
```suggestion
          <div className="error-div">
```

File: app/lib/theme/FieldTemplate.tsx [hunk 6]
```diff
@@ -1,15 +1,38 @@
...
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faExclamationTriangle } from "@fortawesome/free-solid-svg-icons";
+
+const FieldTemplate = ({ children, errors, help, rawErrors }) => {
   return (
     <>
       <div>
         {children}
-        {errors}
+        {rawErrors ? (
```
Comment (matthieu-foucault) [inline app/lib/theme/FieldTemplate.tsx:9]: this condition can be inside of the `error-div`

File: app/lib/theme/FieldTemplate.tsx [hunk 7]
```diff
@@ -1,15 +1,38 @@
...
+          <div className={"error-div"}>
+            <FontAwesomeIcon
+              className="error-icon"
+              icon={faExclamationTriangle}
+            />
+            {errors}
```
Comment (matthieu-foucault) [inline app/lib/theme/FieldTemplate.tsx:15]: There's a bit too much margin between the triangle and the error message

File: app/lib/theme/FieldTemplate.tsx [hunk 8]
```diff
@@ -1,15 +1,38 @@
...
+          </div>
+        ) : (
+          <div className={"error-div"}>
+            <div>&nbsp;</div>
```
Comment (matthieu-foucault) [inline app/lib/theme/FieldTemplate.tsx:19]: is this needed?

File: app/data/jsonSchemaForm/customFormats.ts [hunk 9]
```diff
@@ -0,0 +1,7 @@
+export const customFormats = {
+  rfpDigits: "d{3,4}",
+};
+
+export const customFormatsErrorMessages = {
+  rfpDigits: "random RFP digits should be 3 or 4 digits",
```
Comment (matthieu-foucault) [inline app/data/jsonSchemaForm/customFormats.ts:6]:
```suggestion
  rfpDigits: "Please enter 3 or 4 digits for the random RFP digits",
```

Nicer phrasing :)

**Review by matthieu-foucault — CHANGES_REQUESTED** [review]

File: app/lib/theme/customTransformErrors.ts (hunk 2 above)
Comment (dleard) [reply app/lib/theme/customTransformErrors.ts]: I fiddled with getting the input type into the error object, but didn't find a quick solution. Leaving it as "Please enter a value" for all "required" error messages.

**Review by matthieu-foucault — CHANGES_REQUESTED** [review]
Looks like you just need an update of your e2e and unit tests to match the latest changes

**Review by matthieu-foucault — APPROVED** [review]

**Review by matthieu-foucault — APPROVED** [review]

**Pull request merged**